contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -460,6 +483,28 @@ public void setValue(PointerType key, PointerType value) {
      * the characteristics and behavior of {@code CFString} objects.
      */
     class CFStringRef extends CFTypeRef {
+
+        /**
+         * Placeholder for a reference to a {@code CFString} object.
+         */
+        public static class ByReference extends PointerByReference {
+            public ByReference() {
+            }
+
+            public ByReference(CoreFoundation.CFStringRef value) {
+                super(value.getPointer());
+            }
+
+            public CoreFoundation.CFStringRef getStringRefValue() {
```

Comment:
Something's not sitting right with me on this particular implementation.  I've typed and backspaced multiple comments here but the gist of my concern is that this method can potentially throw a `ClassCastException` if you have the wrong type and it would be better to throw the exception on instantiation or update with an invalid value.

This isn't a problem using the constructor that takes a `CFStringRef` argument, but by extending `PointerByReference` you gain access to `setValue(Pointer value)` so you'd have to override that to add a type check, which seems overly complex.  I'm thinking internally storing a `CFStringRef` and overriding the set/get value methods to use that class methods may be preferred.

Reply:
From my point of view, `setValue` is seldom used, so it's better just to check type id on pointer, see fixed after review